src/platform/plugins/shared/unified_doc_viewer/public/hooks/use_field_types.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import { useMemo } from 'react';
+import type { DataView, DataViewField } from '@kbn/data-views-plugin/common';
+import type { DataTableColumnsMeta } from '@kbn/discover-utils/types';
+import { getDataViewFieldOrCreateFromColumnMeta } from '@kbn/data-view-utils';
+
+interface UseFieldTypesProps<T extends object> {
+  doc: T;
+  dataView: DataView;
+  columnsMeta?: DataTableColumnsMeta;
+}
+
+export const useFieldTypes = <T extends object>({
+  doc,
+  dataView,
+  columnsMeta,
+}: UseFieldTypesProps<T>): { docFieldTypes: Record<string, DataViewField | undefined> } => {
+  const docFieldTypes = useMemo(
+    () =>
+      Object.keys(doc).reduce((acc, fieldName) => {
```

Comment:
Hi @Bluefinger,

Customer indices can include thousands of fields. For better performance it's important to avoid iterating though all of them otherwise it would slow the rendering. Can we instead of this hook rather call `getDataViewFieldOrCreateFromColumnMeta` only for fields which are relevant?